pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+        }
+      }
+      if (atts.size() > 0) {
+        s.setAttributes(atts);
+      }
+
+      return s;
+    }
+
+  }
+
+  private static class TypeRowMapper implements RowMapper<Type> {
+
+    @Override
+    public Type mapRow(ResultSet rs, int rowNum) throws SQLException {
+      Type t = new DefaultType();
+
+      String misoType = rs.getString("miso_type");
+      if ("Sample".equals(misoType)) {
+        t.setName(SampleTypeConverter.getSampleType(rs.getString("name")));
+      } else {
+        t.setName(SampleTypeConverter.getNonSampleSampleType(misoType, rs.getString("sampleType_platform"),
+            rs.getString("sampleType_description")));
+      }
+      t.setCount(rs.getInt("count"));
+      t.setArchivedCount(rs.getInt("archivedCount"));
+      t.setEarliest(rs.getTimestamp("earliest"));
+      t.setLatest(rs.getTimestamp("latest"));
+
+      return t;
+    }
+
+  }
+
+  private static class SampleProjectMapper implements RowMapper<SampleProject> {
+
+    @Override
+    public SampleProject mapRow(ResultSet rs, int rowNum) throws SQLException {
+      SampleProject p = new DefaultSampleProject();
+
+      p.setName(rs.getString("name"));
+      p.setCount(rs.getInt("count"));
+      p.setArchivedCount(rs.getInt("archivedCount"));
+      p.setEarliest(rs.getTimestamp("earliest"));
+      p.setLatest(rs.getTimestamp("latest"));
+
+      return p;
+    }
+
+  }
+
+  private static class ChangeMapper implements RowMapper<MisoChange> {
+
+    @Override
+    public MisoChange mapRow(ResultSet rs, int rowNum) throws SQLException {
+      MisoChange c = new MisoChange();
+
+      c.setAction(rs.getString("action"));
+      c.setCreated(rs.getTimestamp("changeTime"));
+      c.setCreatedById(rs.getInt("userId"));
+      c.setSampleId(rs.getString("sampleId"));
+
+      return c;
+    }
+
+  }
+
+  private static class IdListMapper implements RowMapper<String> {
+
+    @Override
+    public String mapRow(ResultSet rs, int rowNum) throws SQLException {
+      return rs.getString("id");
+    }
+
+  }
+
+  private static final RowMapper<MisoBoxPosition> boxPositionRowMapper = new RowMapper<MisoBoxPosition>() {
+
+    @Override
+    public MisoBoxPosition mapRow(ResultSet rs, int rowNum) throws SQLException {
+      MisoBoxPosition pos = new MisoBoxPosition();
+      pos.setId(rs.getLong("boxId"));
+      pos.setName(rs.getString("alias"));
+      pos.setDescription(rs.getString("description"));
+      pos.setLocation(rs.getString("locationBarcode"));
+      pos.setRows(rs.getInt("rows"));
+      pos.setColumns(rs.getInt("columns"));
+      pos.setPosition(rs.getString("position"));
+      pos.setSampleId(rs.getString("targetType"), rs.getLong("targetId"));
+      return pos;
+    }
+
+  };
+
+  private static String getResourceAsString(String resourceName) {
+    try {
+      return IOUtils.toString(MisoClient.class.getResourceAsStream(resourceName), "UTF-8");
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)